schema_salad/python_codegen_support.py

```diff
@@ -43,6 +43,9 @@
 IdxType = MutableMapping[str, Tuple[Any, "LoadingOptions"]]
 
 
+doc_line_info = CommentedMap()
```

Comment:
Instead of `doc_line_info` being a global variable, how about having `save()` take `LoadingOptions` and using `original_doc` ?